src/useBaseFetch.ts

```diff
@@ -14,19 +14,36 @@ export default <Data>(config: AxiosRequestConfig): BaseFetch<Data> => {
   const [loading, setLoading] = useState(false);
   const [error, setError] = useState<Error | null>(null);
   const isMounted = useRef(true);
-  const { current } = isMounted;
+
+  const { CancelToken } = axios;
+  const source = CancelToken.source();
+
+  const configWithCancelToken = rawConfig => {
+    if (typeof rawConfig === 'string') {
+      return {
+        url: rawConfig,
+        cancelToken: source.token,
+      };
+    }
+    return {
+      ...rawConfig,
+      cancelToken: source.token,
+    };
+  };
 
   const getData = async () => {
     try {
       setLoading(true);
-      const res = (await axios(config)) as AxiosResponse<Data>;
-      if (current) {
+      const res = (await axios(configWithCancelToken(config))) as AxiosResponse<
```

Comment:
This function name is a little confusing - maybe we should name it something like `getConfig`?

Reply:
Changed 👍